minetest/minetest
# Deprecate wrong __index usage in core.register_entity

This PR set the __index attribute to deprecated for the builtin item entity.

The attribute was meant nicely to simplify enhancements, but have the limitation of just 1 enhancement possible. Since the minetest_game/default mod uses the enhancement, no other mods can use them. 
See https://github.com/minetest/minetest_game/pull/2328#issuecomment-531285816 and the rest of minetest game pull request for more technical background around the issue.

The first proposal was just to remove the `prototype.__index = prototype` after the change https://github.com/minetest/minetest_game/pull/2328/files is merged to the minetest game, but p_gimeno has pointed up we should keep it because maybe used in other games.

My proposal now is to keep the compatibility code from this PR for 1 release period with the deprication warning and then remove all the lines.

How to test:
1 Merge this PR and try to run Minetest Game - Warning appears if first item is dropped to the world:
```
2020-02-17 08:53:19: WARNING[Server]: Entity __builtin:item from mod default uses deprecated setmetatable(item, builtin_item) method
2020-02-17 08:53:19: WARNING[Server]: See https://github.com/minetest/minetest_game/pull/2328/ for details
```
Merge the https://github.com/minetest/minetest_game/pull/2328 to Minetest game
-> Game still works as before, but without the warning
